## Review: Bounce-Back Cycle 1

**Reason:** CI failed (unrelated, will rebase)

**Reviewer A (Correctness):** APPROVE — Tuple reorder + caller updates correctly invert drop order so `CatchupManager` drops before its backing `TempDir`. All three call sites updated; no other callers.
**Reviewer B (Risk):** APPROVE — Test-only, `mod tests`-private helper tuple reorder; zero production surface, zero API/data-format/security/perf impact, drop-order change is strictly safer.
**External reviewers:** copilot-pull-request-reviewer[bot]: COMMENTED (non-blocking).
**CI:** red (unrelated)

The two failing checks are `build / 3 test / 2 test (testing-with-pr, amd64, local, rpc , )` and `build / 3 test / 2 test (testing-with-pr, amd64, local, core,rpc,horizon , )`, plus their rollup jobs (`build / 3 test / complete`, `build / complete`). Both timed out in the **Quickstart** Docker integration tests waiting for `stellar-rpc` / `friendbot` to come up:

```
stellar-rpc: waiting for ready state, 4 minutes...
supervisor: WARN exited: stellar-rpc (exit status 1; not expected)
[test_friendbot] Waiting for Friendbot to be available
##[error]The action 'Run friendbot test' has timed out after 4 minutes.
```

The PR's diff only touches `crates/history/src/catchup/replay.rs` inside `mod tests` (tuple element reorder in a private test helper) — it cannot affect Docker / `stellar-rpc` / `friendbot` startup. Recent Quickstart runs on `main` are also failing (run IDs 25952505373, 25951830752), confirming this is an infra/upstream flake, not diff-attributable.

Routing back to `ready-for-doing` for `/do` Mode B to rebase on `origin/main` and re-push. If CI is still red after rebase, the next `/review-pr` will mark this `blocked`.

--- a/crates/history/src/catchup/replay.rs
+++ b/crates/history/src/catchup/replay.rs
@@ -817,11 +817,16 @@ mod tests {
     //     `apply_data[0]` for the case-4 (Apply) check.
     // ---------------------------------------------------------------
 
-    /// Returns the manager together with the `TempDir` guard backing its
-    /// `BucketManager`. The caller must keep the `TempDir` alive for the
-    /// duration of the test so the bucket directory is deleted on drop
-    /// rather than leaked under the system temp location.
-    fn make_test_catchup_manager() -> (CatchupManager, tempfile::TempDir) {
+    /// Returns the `TempDir` guard backing the `BucketManager` together
+    /// with the manager. The `TempDir` is returned **first** so callers
+    /// destructure as `let (_tmp_dir, manager) = ...`: Rust drops local
+    /// bindings in reverse declaration order, so binding `manager`
+    /// second guarantees it (and any file handles its `BucketManager`
+    /// holds) is dropped before the `TempDir` removes the directory.
+    /// The caller must keep `_tmp_dir` alive for the duration of the
+    /// test so the bucket directory is deleted on drop rather than
+    /// leaked under the system temp location.
+    fn make_test_catchup_manager() -> (tempfile::TempDir, CatchupManager) {
         use henyey_bucket::BucketManager;
         use henyey_db::Database;
 
@@ -831,8 +836,8 @@ mod tests {
             BucketManager::new(tmp_dir.path().to_path_buf()).expect("bucket manager");
         let archive = crate::HistoryArchive::new("https://example.com").expect("archive");
         (
-            CatchupManager::new(vec![archive], bucket_manager, db),
             tmp_dir,
+            CatchupManager::new(vec![archive], bucket_manager, db),
         )
     }
 
@@ -862,7 +867,7 @@ mod tests {
     fn test_verify_knit_to_lcl_happy_path_mixed() {
         // LCL = 100, lcl_hash = H100, previous_ledger_hash = H99.
         let lcl = make_test_lcl(100, h(0x10), h(0x09));
-        let (manager, _tmp_dir) = make_test_catchup_manager();
+        let (_tmp_dir, manager) = make_test_catchup_manager();
 
         // Three knit entries spanning cases 1/2/3 — each must classify
         // as Skip inside the loop, exercising 3 iterations of the body.
@@ -896,7 +901,7 @@ mod tests {
     fn test_verify_knit_to_lcl_tampered_knit_entry_returns_fatal() {
         // LCL = 100, lcl_hash = H100, previous_ledger_hash = H99.
         let lcl = make_test_lcl(100, h(0x10), h(0x09));
-        let (manager, _tmp_dir) = make_test_catchup_manager();
+        let (_tmp_dir, manager) = make_test_catchup_manager();
         // apply_data is irrelevant — the loop must error out before
         // we reach the `.first()` branch.
         let apply_data = vec![make_apply_ledger_data(101, h(0x10))];
@@ -938,7 +943,7 @@ mod tests {
     #[test]
     fn test_verify_knit_to_lcl_empty_apply_data_noop() {
         let lcl = make_test_lcl(100, h(0x10), h(0x09));
-        let (manager, _tmp_dir) = make_test_catchup_manager();
+        let (_tmp_dir, manager) = make_test_catchup_manager();
 
         // No knit entries, no apply data: pure no-op, must Ok.
         manager

